src/main/java/org/junit/runners/ParentRunner.java

```diff
@@ -187,12 +187,14 @@ private void validateClassRules(List<Throwable> errors) {
      * @return {@code Statement}
      */
     protected Statement classBlock(final RunNotifier notifier) {
-        Statement statement = childrenInvoker(notifier);
-        if (!areAllChildrenIgnored()) {
-            statement = withBeforeClasses(statement);
-            statement = withAfterClasses(statement);
-            statement = withClassRules(statement);
+        if (areAllChildrenIgnored()) {
+            return withIgnores(null, notifier);
         }
+        Statement statement = childrenInvoker(notifier);
+        statement = withBeforeClasses(statement);
+        statement = withIgnores(statement, notifier);
```

Comment:
shouldn't this be the last thing you do before returning the statement chain? If the class rules or `@AfterClass` methods throw, don't you want the ignored methods to be notified?